pan3d/ui/render_area.py

```diff
@@ -0,0 +1,15 @@
+from trame.widgets import html, vuetify
+
+
+class RenderArea(html.Div):
+    def __init__(self, plot_view=None):
+        super().__init__(
+            v_show="array_active",
+            style="height: 100%; position: relative; width: calc(100% - 300px)",
+        )
+        with self:
+            vuetify.VBanner(
+                "{{ error_message }}",
+                v_show=("error_message",),
+            )
+            plot_view
```

Comment:
As far as I know like 15 is a NoOp. Are you expecting something from that?

Reply:
I was hoping I could do that because I had to move the plotter creation back into the main class since it requires the controller. Since passing it through doesn't work, I'll just dissolve this class back into the main one. This class can't have any reason to exist if it's not doing the job of bringing the plotter ui into the layout.

Reply:
Well you can do it but not like that.

Reply:
See 

```python
                    
with RenderArea() as view:
    with view.content:
        plot_view = plotter_ui(
            self.ctrl.get_plotter(),
            interactive_ratio=1,
        )
        self.ctrl.view_update = plot_view.update
        self.ctrl.reset_camera = plot_view.reset_camera


```

Reply:
Think of what you do in vue.js with slots... This is the same concept.

Reply:
I see. Do you think it's still worth separating a component for it?

Reply:
If it is tiny, probably not... But that split is up to you to decide what make sense.

Here is an example of one that I just made

```python
class VCard(vuetify.VCard):
    def __init__(self, show_name, title=None, **kwargs):
        super().__init__(**kwargs)
        with self:
            self.header = vuetify.VCardTitle(title, classes="py-2")
            with self.header:
                vuetify.VSpacer()
                vuetify.VIcon(
                    v_text=f"{show_name} ? 'mdi-chevron-up':'mdi-chevron-down'",
                    click=f"{show_name} = !{show_name}",
                )
            vuetify.VDivider(v_show=show_name)
            self.content = vuetify.VCardText(v_show=(show_name, True))

```

Reply:
I'll leave it in the main component for now, since we moved the UI creation to the lazily-loaded viewer function. It's pretty tiny as just the display of the error and the plotter_ui object from pyvista, so it doesn't warrant its own file. Having the other three components separated makes the viewer code a reasonable size.

Reply:
That was the intent so we could have a grasp of the big picture of what the viewer looks like (drawer, content, ...)